src/jvm/main/org/jetbrains/kotlinx/lincheck/runner/ParallelThreadsRunner.kt

```diff
@@ -307,6 +302,8 @@ internal open class ParallelThreadsRunner(
             // Execute the parallel part.
             beforePart(PARALLEL)
             timeout -= executor.submitAndAwait(parallelPartExecutions, timeout)
```

Comment:
Yes. 

- `executor.submitAndAwait(parallelPartExecutions, timeout)` submits the tasks and awaits for their compeltion. 
- `strategy.awaitAllThreads(timeout)` awaits for the completion of all threads registered in the strategy (including user custom threads).

The only reason why `strategy.awaitAllThreads(timeout)` is needed now is because now we also have custom threads, so we also need to wait for their completion. 

Previously, number of threads was equal to the number of submitted tasks to the executor, so no additional waiting was required after `executor.submitAndAwait(parallelPartExecutions, timeout)`.

Reply:
Can we keep `strategy.awaitAllThreads(timeout)` and remove `executor.submitAndAwait(parallelPartExecutions, timeout)`?

Reply:
We cannot remove `executor.submitAndAwait(parallelPartExecutions, timeout)` --- its both submits and awaits the tasks. We need to at least submit the tasks : )

I can rename `strategy.awaitAllThreads(timeout)` to `strategy.awaitUserThreads(timeout)` to make the code less confusing.

Reply:
Renamed `awaitAllThreads` to `awaitUserThreads`.